yarn-project/noir-contracts/src/contracts/schnorr_multi_key_account_contract/src/storage.nr

```diff
@@ -0,0 +1,21 @@
+use dep::aztec::state_vars::{
+    immutable_singleton::ImmutableSingleton
+};
+
+use crate::public_key_note::{
+    PublicKeyNote,
+    PublicKeyNoteInterface,
+    PUBLIC_KEY_NOTE_LEN,
+};
+
+struct Storage {
+    public_key: ImmutableSingleton<PublicKeyNote, PUBLIC_KEY_NOTE_LEN>,
+}
+
+impl Storage {
+    fn init() -> Self {
+        Storage {
+            public_key: ImmutableSingleton::new(1, PublicKeyNoteInterface)
```

Comment:
I would specify that it's a signing public key. Otherwise it's a bit unclear what kind of key we are talking about.